File: Code Generation/responses/python/gemini/easy/minimum-cost-to-reach-every-position.py
from typing import List

class Solution:
    def minCosts(self, cost: List[int]) -> List[int]:
        ans = []
        current_min = float('inf')
        for c in cost:
            if c < current_min:
                current_min = c
            ans.append(current_min)
        return ans

File: Code Generation/responses/cpp/deepseek/easy/minimum-cost-to-reach-every-position.cpp
#include <vector>
#include <algorithm>

using namespace std;

class Solution {
public:
    vector<int> minCosts(vector<int> cost) {
        int n = cost.size();
        vector<int> answer(n);
        
        int min_cost = cost[0];
        for (int i = 0; i < n; i++) {
            min_cost = min(min_cost, cost[i]);
            answer[i] = min_cost;
        }
        
        return answer;
    }
};